frontend/src/views/instructor/ddahs/ddah-editor.tsx

```diff
@@ -0,0 +1,509 @@
+            newDuty = {
+                ...newDuty,
+                description: `${category}:${newDuty.description}`,
+            };
+            return { ...duty, ...newDuty };
+        });
+        setDuties(newDuties);
+    }
+    function onDutyAdd(category: Category) {
+        const maxOrder = Math.max(...duties.map((duty) => duty.order));
+        const order = Number.isFinite(maxOrder) ? maxOrder + 1 : -1;
+        setDuties([
+            ...duties,
+            { order, hours: 0, description: `${category}:` },
+        ]);
+    }
+    function onDutyDelete(duty: Duty) {
+        setDuties(duties.filter((d) => d.order !== duty.order));
+    }
+
+    if (ddah != null) {
+        let totalHours = 0;
+        for (const duty of duties) {
+            totalHours += duty.hours;
+        }
+        const hoursMismatch = ddah.assignment
+            ? ddah.assignment.hours !== totalHours
+            : false;
+        const dutiesByCategory: Record<Category, Duty[]> = {
+            note: [],
+            prep: [],
+            training: [],
+            meeting: [],
+            contact: [],
+            marking: [],
+            other: [],
+        };
+        for (const duty of duties) {
+            const { category, description } = splitDutyDescription(
+                duty.description
+            ) as { category: Category; description: string };
+            dutiesByCategory[category] = dutiesByCategory[category] || [];
+            dutiesByCategory[category].push({ ...duty, description });
+        }
+
+        const assignment = ddah.assignment;
+        const applicant = assignment.applicant;
+        const position = assignment.position;
+
+        ddahPreview = (
+            <div className="instructor-ddah-preview-container">
+                <table className="position-summary">
+                    <tbody>
+                        <tr>
+                            <th>TA:</th>
+                            <td>
+                                {applicant.first_name} {applicant.last_name}
+                            </td>
+                        </tr>
+                        <tr>
+                            <th>Position:</th>
+                            <td>
+                                {position.position_code} (
+                                {position.position_title})
+                            </td>
+                        </tr>
+                    </tbody>
+                </table>
+                <h4>Notes</h4>
+                {editing && (
+                    <Alert variant="secondary">
+                        <FaInfoCircle className="mr-2" />
+                        {categoryInformation["note"].helpText}
+                    </Alert>
+                )}
+                <ul>
+                    <DutyList
+                        category={"note"}
+                        duties={dutiesByCategory["note"] || []}
+                        showEmptyDutyString={false}
```

Comment:
I see this prop is only used once, is there a use-case where we'd want to use the default `true`?

Reply:
We want it `true` most of the time. Just not for `notes`